WordPress/gutenberg
# Update InputControl and SelectControl default heights

## What?
Make the default `height` / `min-height` for `InputControl` and `SelectControl` `32px` rather than `30px`.

## Why?
Interactive elements like buttons and inputs are trending towards a 24px, 32px, and 40px sizing scale. The current 30px default applied to InputControl and SelectControl violates this convention resulting in awkward sizing mismatches, for example in the data view pagination UI:

<img width="256" alt="Screenshot 2023-10-19 at 15 00 38" src="https://github.com/WordPress/gutenberg/assets/846565/54b0dbea-f2e0-45d0-871e-e79d7e00a42b">

The icon button(s) are 32px tall while the page number input is only 30px. This issue will likely become more apparent as we proceed with the admin redesign project.

## Testing Instructions
1. `npm run storybook:dev`
2. Open Storybook (should happen automatically when the previous command completes)
3. Observe that `InputControl` and `SelectControl` (and any components that utilise them e.g. `UnitControl`) are 32px tall when `size` is `default` and `__next40pxDefaultSize` is `false`.
4. Check instances of these components in the Editor

| Before | After |
| --- | --- |
| <img width="583" alt="Screenshot 2023-10-19 at 15 04 21" src="https://github.com/WordPress/gutenberg/assets/846565/815f00a3-ce33-4d95-aee9-ec92d2109b87"> | <img width="564" alt="Screenshot 2023-10-19 at 15 04 12" src="https://github.com/WordPress/gutenberg/assets/846565/f6e6138d-01cb-4cd8-961a-249c68dc20b3"> |
